src/CoreWCF.Http/tests/Authorization/AuthorizationIntegrationTests.Service.cs

```diff
@@ -10,7 +10,7 @@ namespace CoreWCF.Http.Tests.Authorization;
 public partial class AuthorizationIntegrationTests
 {
     [ServiceContract]
-    private interface ISecuredService
+    internal interface ISecuredService
     {
```

Comment:
I presume that private interfaces don't work with a source generated invoker (which makes sense). What happens in that case? What do you think we should do? Turning on source generation of the invoker is an opt-in feature, so we could fail the build, or we could emit a warning that we couldn't generate the invoker for this type, or we could just not attempt to emit anything. I don't like that last option though as it makes discovery difficult.

Reply:
Current behavior is that we got a CS0122 build error because we do generate the code.
![image](https://github.com/CoreWCF/CoreWCF/assets/681739/88486afe-8ed9-4c02-bc83-f15ba53ad4cd)

I see 3 potential options:
1. do nothing, the build is broken but the compiler error is not explicit about the source generator.
2. detect the case in the source gen (and not emit the OperationInvoker implementation) and emit a build error more meaningful for user.
3. detect the case in the source gen (and not emit the OperationInvoker implementation) and emit a build warning more meaningful for user. (this case will lead to runtime exceptions). 

As this feature mainly target AOT scenario where the difficulty is to track potential runtime issue at build time I would go for either 1. or 2.

Note: The actual behavior of the `EnableCoreWCFOperationInvokerGenerator` project property is that we add a check in `src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/OperationInvokerBehavior.cs` so we trigger a PlatformNotSupportedException if the feature is enabled and we cannot find the implementation in the dictionary that contains registered implementations. We may also have more finegrained behavior and introduce a fallback towards the current non generated OperationInvoker implemenation. This finegrained behavior could be controlled by having values like "Always" / "IfPossible" / "Never" in `EnableCoreWCFOperationInvokerGenerator`. But i'm not sure we actually need this.

Reply:
I believe the original code will work with aot, it just has bad performance characteristics. The DataContractSerializer has to use MethodInfo.Invoke. I don't like the idea of a full break unless there's a way to override it. There are a few reasons why that might be okay. It could be a contract interface only used by a WCF client so doesn't need an invoker generated. It could be an interface in a shared library which isn't used to expose an endpoint in the current project.  

Could we make it an analyzer which creates an error at the point of use of the contract interface? That allows the developer to put in an ignore override in the code with a disable pragma, or even at the project level. This means if the interface isn't used anywhere, no error and it's silently ignored. We can't just worry about use with AddServiceEndpoint though as I've seen developers use a generic class to host multiple services/contracts so AddServiceEndpoint might not have the contract type at the call point.

Reply:
> I believe the original code will work with aot, it just has bad performance characteristics. The DataContractSerializer has to use MethodInfo.Invoke. I don't like the idea of a full break unless there's a way to override it. There are a few reasons why that might be okay. It could be a contract interface only used by a WCF client so doesn't need an invoker generated. It could be an interface in a shared library which isn't used to expose an endpoint in the current project.
> 
> Could we make it an analyzer which creates an error at the point of use of the contract interface? That allows the developer to put in an ignore override in the code with a disable pragma, or even at the project level. This means if the interface isn't used anywhere, no error and it's silently ignored. We can't just worry about use with AddServiceEndpoint though as I've seen developers use a generic class to host multiple services/contracts so AddServiceEndpoint might not have the contract type at the call point.

I updated the analyzer according to this requirement. We nos output a build error at the location the interface is used if the interface is private.